api/src/org/apache/cloudstack/api/command/user/vm/ListNicsCmd.java

```diff
@@ -122,22 +123,44 @@ public static String getResultObjectName() {
     public void execute() throws ResourceUnavailableException, ResourceAllocationException, ConcurrentOperationException, InsufficientCapacityException {
 
         try {
-            List<? extends Nic> results = _networkService.listNics(this);
-            ListResponse<NicResponse> response = new ListResponse<NicResponse>();
-            List<NicResponse> resList = null;
-            if (results != null) {
-                resList = new ArrayList<NicResponse>(results.size());
-                for (Nic r : results) {
-                    NicResponse resp = _responseGenerator.createNicResponse(r);
-                    resp.setObjectName("nic");
-                    resList.add(resp);
+            if (this.getKeyword() != null && !this.getKeyword().isEmpty() && this.getNicId() != null) {
+                List<? extends NicSecondaryIp> results = _networkService.listSecondaryNics(this);
```

Comment:
Change the method name listSecondaryNics.  Here method is not listing the secondary nics it is listing nic secondary ips. The method name some thing like listVmNicSecondaryIps